Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/SurfaceMagnetism.cs

```diff
@@ -7,96 +7,253 @@
 namespace Microsoft.MixedReality.Toolkit.Utilities.Solvers
 {
     /// <summary>
-    /// SurfaceMagnetism casts rays to Surfaces in the world align the object to the surface.
+    /// SurfaceMagnetism casts rays to Surfaces in the world and aligns the object to the hit surface.
     /// </summary>
     public class SurfaceMagnetism : Solver
     {
-        private const float MaxDot = 0.97f;
-
-        private enum RaycastDirectionEnum
+        #region Enums
+        public enum RaycastDirectionType
         {
-            CameraFacing,
+            CameraFacing = 0,
             ToObject,
             ToLinkedPosition
         }
 
-        private enum OrientModeEnum
+        public enum OrientionMode
```

Comment:
I *think* that this isn't breaking because of how Unity encodes the enum values (as ints) in the serialized files, so I think this should be fine.

Note that this is misspelled however:

OrientionMode -> OrientationMode

Reply:
@wiwei Correct, as long as the field name is the same it will be OK.

Reply:
yes it is not a breaking change but I am idiot for not knowing how to spell :P